docs/guide/api-hmr.md

```diff
@@ -147,7 +147,15 @@ if (import.meta.hot) {
 
 ## `hot.data`
 
-The `import.meta.hot.data` object is persisted across different instances of the same updated module. It can be used to pass on information from a previous version of the module to the next one.
+The `import.meta.hot.data` object is persisted across different instances of the same updated module. It can be used to pass on information from a previous version of the module to the next one. Note that re-assignment of `data` itself is not supported. Instead, you can mutate properties of `data` object.
```

Comment:
```suggestion
The `import.meta.hot.data` object is persisted across different instances of the same updated module. It can be used to pass on information from a previous version of the module to the next one. 

Note that re-assignment of `data` itself is not supported. Instead, you should mutate properties of the `data` object so information added from other handlers are preserved.
```

I think we can add extra context for why it's not supported, otherwise they might think it could be supported in the future. Also added an extra line for breathability.